test/ci-cd/scripts/terraservice/apply_initialize.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source /workspace/build.env
+if [ "${DEBUG,,}" == "true" ]; then
+  set -o xtrace
+fi
+
+MY_PATH="$(
+  cd "$(dirname "$0")" >/dev/null 2>&1
+  pwd -P
+)"
+
+TERRASERVICE_DIR="${1}"
+TERRASERVICE_FOLDER="${2}"
+
+source "${ACP_PLATFORM_BASE_DIR}/_shared_config/scripts/set_environment_variables.sh" "${ACP_PLATFORM_BASE_DIR}/_shared_config"
+
+export STATE_MIGRATED="false"
+if gcloud storage ls "gs://${terraform_bucket_name}/terraform" &>/dev/null; then
+  export STATE_MIGRATED="true"
+fi
+
```

Comment:
This is not the first time I see this check to set `STATE_MIGRATED`. Consider moving this in a function, and put it in `platforms/gke/base/core/functions.sh`.

Reply:
The migrate state check is just to make the scripts idempotent for our CI/CD and local development. Ideally when you use the repository you should commit your changes after doing the initialization.